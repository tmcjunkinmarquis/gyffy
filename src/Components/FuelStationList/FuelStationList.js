import React from '../../../../../../../Library/Caches/typescript/2.9/node_modules/@types/react';

const FuelStationLIst = ()=>{


}

export