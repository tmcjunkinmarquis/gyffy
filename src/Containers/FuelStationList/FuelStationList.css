h3  {
  color: black;
  text-shadow: none;
}

span {
  color: darkslateblue;
  white-space: pre;
}