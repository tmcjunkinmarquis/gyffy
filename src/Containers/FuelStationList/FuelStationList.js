import React, { Component } from 'react';
import { connect } from 'react-redux';
import './FuelStationList.css';
import StationCard from '../../Components/StationCard/StationCard';
import PropTypes from 'prop-types';
import { setFilter } from '../../Actions/Actions';

class FuelStationList extends Component {
  
  componentDidMount(){
    this.props.handleFilterClick('SHOW_ALL')
  }
  
  makeAstation = ()=>{
    return this.props.stations.map((station)=>{
      return <StationCard 
        station={station}
        key={station.id}
        fuelType={station.fuelType}/>;
    });
  };

  fillTheSpan = () => {
    return 'ALL  '
  }

  render(){
    return (
      <div className="wrapper">
        <h3 className="fuel-stations-header"><span className="filter-type">{this.fillTheSpan()}</span>STATIONS
          <span >   -   {this.props.location}</span>
        </h3>
        <div className="fuel-station-list">{this.makeAstation()}</div>
      </div>
    );
  } 
}

export const mapStateToProps = (state)=>({
  location: state.location,
  stations: state.stations,
  filter: state.filter
});

export const mapDispatchToProps = (dispatch) => ({
  handleFilterClick: (filter) => dispatch(setFilter(filter))
});

export default connect(mapStateToProps, mapDispatchToProps)(FuelStationList);

FuelStationList.propTypes = {
  stations: PropTypes.array
};