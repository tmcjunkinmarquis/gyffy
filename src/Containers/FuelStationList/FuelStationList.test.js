import React from 'react';
import { shallow } from 'enzyme';
import { FuelStationList, mapStateToProps, mapDispatchToProps } from './FuelStationList';

describe('FuelStationList Container', () => {
  

  describe('FuelStationList Component', () => {
    
    let wrapper;
    let mockProps;
    let mockHandleFilterClick;
    beforeEach(()=>{
      mockHandleFilterClick = jest.fn()
      mockProps = {
        stations: [{ name: 'bubbas' }],
        location: '61615',
        filter: 'SHOW_ELEC'
      }
      wrapper = shallow(
        <FuelStationList 
          handleFilterClick = {mockHandleFilterClick}
          {...mockProps}/>)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Name of the group', () => {
    
  });
});