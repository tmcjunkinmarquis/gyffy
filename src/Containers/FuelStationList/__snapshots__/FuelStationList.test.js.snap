// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FuelStationList Container FuelStationList Component should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FuelStationList
    filter="SHOW_ELEC"
    handleFilterClick={[Function]}
    location="61615"
    stations={
        Array [
            Object {
              "name": "bubbas",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="fuel-stations-header"
>
          <span
                    className="filter-type"
          >
                    ALL  
          </span>
          STATIONS
          <span>
                       -   
                    61615
          </span>
</h3>,
        <div
          className="fuel-station-list"
>
          <Link
                    replace={false}
                    to="/details/undefined"
          >
                    <Connect(Station)
                              fuelType={undefined}
                              station={
                                        Object {
                                                  "name": "bubbas",
                                                }
                              }
                    />
          </Link>
</div>,
      ],
      "className": "wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="filter-type"
>
              ALL  
</span>,
            "STATIONS",
            <span>
                 -   
              61615
</span>,
          ],
          "className": "fuel-stations-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ALL  ",
              "className": "filter-type",
            },
            "ref": null,
            "rendered": "ALL  ",
            "type": "span",
          },
          "STATIONS",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "   -   ",
                "61615",
              ],
            },
            "ref": null,
            "rendered": Array [
              "   -   ",
              "61615",
            ],
            "type": "span",
          },
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/details/undefined"
>
              <Connect(Station)
                            fuelType={undefined}
                            station={
                                          Object {
                                                        "name": "bubbas",
                                                      }
                            }
              />
</Link>,
          ],
          "className": "fuel-station-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(Station)
                fuelType={undefined}
                station={
                                Object {
                                                "name": "bubbas",
                                              }
                }
/>,
              "replace": false,
              "to": "/details/undefined",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "fuelType": undefined,
                "station": Object {
                  "name": "bubbas",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="fuel-stations-header"
>
            <span
                        className="filter-type"
            >
                        ALL  
            </span>
            STATIONS
            <span>
                           -   
                        61615
            </span>
</h3>,
          <div
            className="fuel-station-list"
>
            <Link
                        replace={false}
                        to="/details/undefined"
            >
                        <Connect(Station)
                                    fuelType={undefined}
                                    station={
                                                Object {
                                                            "name": "bubbas",
                                                          }
                                    }
                        />
            </Link>
</div>,
        ],
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="filter-type"
>
                ALL  
</span>,
              "STATIONS",
              <span>
                   -   
                61615
</span>,
            ],
            "className": "fuel-stations-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ALL  ",
                "className": "filter-type",
              },
              "ref": null,
              "rendered": "ALL  ",
              "type": "span",
            },
            "STATIONS",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "   -   ",
                  "61615",
                ],
              },
              "ref": null,
              "rendered": Array [
                "   -   ",
                "61615",
              ],
              "type": "span",
            },
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/details/undefined"
>
                <Connect(Station)
                                fuelType={undefined}
                                station={
                                                Object {
                                                                "name": "bubbas",
                                                              }
                                }
                />
</Link>,
            ],
            "className": "fuel-station-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Connect(Station)
                  fuelType={undefined}
                  station={
                                    Object {
                                                      "name": "bubbas",
                                                    }
                  }
/>,
                "replace": false,
                "to": "/details/undefined",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "fuelType": undefined,
                  "station": Object {
                    "name": "bubbas",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
